### PR TITLE
Fix(hosting): fix and improve ovhConfig loading

### PR DESCRIPTION
**Pull request by rfougier**

### CODE REVIEW

**Comment by rfougier** [comment]
⚠️  Controller will be reviewed in future PR

File: client/app/hosting/hosting.controller.js [hunk 1]
```diff
@@ -60,63 +60,53 @@ angular
 
         function loadOvhConfig () {
             HostingOvhConfig.get($stateParams.productId).then((ovhConfig) => {
```
Comment (FredericEspiau) [inline client/app/hosting/hosting.controller.js]: `return`

File: client/app/hosting/hosting.controller.js [hunk 2]
```diff
@@ -60,63 +60,53 @@ angular
...
-                            loadOvhConfig();
-                        });
-                    } else if ($scope.ovhConfig) {
-                        $scope.ovhConfig.taskPending = false;
-                    }
-                });
-
-                HostingTask.getError($stateParams.productId).then((tasks) => {
-                    if ($scope.ovhConfig) {
+                HostingTask.getPending($stateParams.productId)
```
Comment (FredericEspiau) [inline client/app/hosting/hosting.controller.js]: You could merge the promises:

`.then(() => HostingTask.getPending(...`

File: client/app/hosting/hosting.controller.js [hunk 3]
```diff
@@ -60,63 +60,53 @@ angular
...
+                    .then((tasks) => {
+                        let queue;
                         if (tasks && tasks.length > 0) {
```
Comment (FredericEspiau) [inline client/app/hosting/hosting.controller.js]: `_(tasks).isArray() && !_(tasks).isEmpty()`

File: client/app/hosting/hosting.controller.js [hunk 4]
```diff
@@ -60,63 +60,53 @@ angular
...
+                            const taskPendingMessage = $scope.i18n[`hosting_global_php_version_pending_task_${tasks[0].function.replace(/ovhConfig\//, "")}`];
+                            _.set($scope.ovhConfig, "taskPending", taskPendingMessage || $scope.i18n.hosting_global_php_version_pending_task_common);
+
+                            queue = _.map(tasks, (task) =>
+                                HostingTask.poll($stateParams.productId, task).catch(() => {
+                                    _.set($scope.ovhConfig, "taskPendingError", false);
+                                })
+                            );
+
+                            $q.all(queue).then(() => {
```
Comment (FredericEspiau) [inline client/app/hosting/hosting.controller.js]: `return` but this should be in a service

**Pull request merged**